core/trino-main/src/main/resources/webapp/src/components/LivePlan.jsx

```diff
@@ -246,23 +244,27 @@ export class LivePlan extends React.Component<LivePlanProps, LivePlanState> {
                 graph.setEdge("node-" + source, nodeId, {class: "plan-edge", arrowheadClass: "plan-arrowhead"});
             });
 
-            if (node.remoteSources.length > 0) {
-                graph.setNode(nodeId, {label: '', shape: "circle"});
-
-                node.remoteSources.forEach(sourceId => {
-                    const source = allStages.get(sourceId);
-                    if (source) {
-                        const sourceStats = source.stageStats;
-                        graph.setEdge("stage-" + sourceId + "-root", nodeId, {
+            var sourceFragmentIds = node.descriptor['sourceFragmentIds'];
```

Comment:
This seems a bit fragile, we're relying on descriptor Map for RemoteSourceNode to contain an entry with `sourceFragmentIds` key. That could easily get modified in PlanPrinter, so at a minimum we would need to guard against it with a test.
Would it make sense to have different JsonRenderedNode for web UI and EXPLAIN to get rid of remoteSources in EXPLAIN ? @martint wdyt ?

Reply:
I've added validation test for `sourceFragmentIds`.